plugins/chat/app/models/chat_message.rb

```diff
@@ -226,8 +226,27 @@ def url
     "/chat/message/#{self.id}"
   end
 
+  def create_mentions(user_ids)
+    User.where(id: user_ids).each { |user| ChatMention.create!(chat_message: self, user: user) }
+  end
+
+  def update_mentions(mentioned_user_ids)
+    old_mentions = chat_mentions.pluck(:user_id)
+    updated_mentions = mentioned_user_ids
+    mentions_to_drop = old_mentions - updated_mentions
+    mentions_to_add = updated_mentions - old_mentions
+
+    delete_mentions(mentions_to_drop)
+    create_mentions(mentions_to_add)
+  end
+
   private
 
+  def delete_mentions(user_ids)
+    mentions_to_drop = chat_mentions.filter { |m| user_ids.include?(m.user.id) }
+    chat_mentions.delete(mentions_to_drop)
```

Comment:
Yes, that'll be better.

And I realized that I actually had to use `destroy` instead of `delete` (if we delete a `chat_mention`, an associated `notification` should be deleted too, and in case of delete that wouldn't happen). So I changed it to `destroy_all`.